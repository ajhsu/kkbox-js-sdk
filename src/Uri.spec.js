const { Territory } = require('./constants');
const Uri = require('./Uri');

describe('Uri', () => {
  describe('Auth', () => {
    test('fetchAccessToken', () => {
      expect(Uri.Auth.fetchAccessToken()).toMatchSnapshot();
    });
  });

  describe('Track', () => {
    test('fetch', () => {
      // Use-cases
      expect(
        Uri.Track.fetch('fakeTrackId', {
          territory: Territory.TW
        })
      ).toMatchSnapshot();

      // Exceptions
      expect(() => Uri.Track.fetch()).toThrow();
      expect(() => Uri.Track.fetch('fakeTrackId')).toThrow();
    });
  });

  describe('Album', () => {
    test('fetchMetadata', () => {
      // Use-cases
      expect(
        Uri.Album.fetchMetadata('fakeAlbumId', {
          territory: Territory.TW
        })
      ).toMatchSnapshot();

      // Exceptions
      expect(() => Uri.Album.fetchMetadata()).toThrow();
      expect(() => Uri.Album.fetchMetadata('fakeAlbumId')).toThrow();
    });

    test('fetchTracks', () => {
      // Use-cases
      expect(
        Uri.Album.fetchTracks('fakeAlbumId', {
          territory: Territory.TW,
          limit: 5,
          offset: 10
        })
      ).toMatchSnapshot();

      // Exceptions
      expect(() => Uri.Album.fetchTracks()).toThrow();
      expect(() => Uri.Album.fetchTracks('fakeAlbumId')).toThrow();
    });
  });

  describe('Artist', () => {
    test('fetchMetadata', () => {
      // Use-cases
      expect(
        Uri.Artist.fetchMetadata('fakeArtistId', {
          territory: Territory.TW
        })
      ).toMatchSnapshot();

      // Exceptions
      expect(() => Uri.Artist.fetchMetadata()).toThrow();
      expect(() => Uri.Artist.fetchMetadata('fakeArtistId')).toThrow();
    });

    test('fetchAlbums', () => {
      // Use-cases
      expect(
        Uri.Artist.fetchAlbums('fakeArtistId', {
          territory: Territory.TW,
          limit: 5,
          offset: 10
        })
      ).toMatchSnapshot();

      // Exceptions
      expect(() => Uri.Artist.fetchAlbums()).toThrow();
      expect(() => Uri.Artist.fetchAlbums('fakeArtistId')).toThrow();
    });

    test('fetchTopTracks', () => {
      // Use-cases
      expect(
        Uri.Artist.fetchTopTracks('fakeArtistId', {
          territory: Territory.TW,
          limit: 5,
          offset: 10
        })
      ).toMatchSnapshot();

      // Exceptions
      expect(() => Uri.Artist.fetchTopTracks()).toThrow();
      expect(() => Uri.Artist.fetchTopTracks('fakeArtistId')).toThrow();
    });

    test('fetchRelatedArtists', () => {
      // Use-cases
      expect(
        Uri.Artist.fetchRelatedArtists('fakeArtistId', {
          territory: Territory.TW,
          limit: 5,
          offset: 10
        })
      ).toMatchSnapshot();

      // Exceptions
      expect(() => Uri.Artist.fetchRelatedArtists()).toThrow();
      expect(() => Uri.Artist.fetchRelatedArtists('fakeArtistId')).toThrow();
    });
  });

  describe('Playlist', () => {
    test('fetch', () => {
      // Use-cases
      expect(
        Uri.Playlist.fetch('fakePlaylistId', {
          territory: Territory.TW
        })
      ).toMatchSnapshot();

      // Exceptions
      expect(() => Uri.Playlist.fetch()).toThrow();
      expect(() => Uri.Playlist.fetch('fakePlaylistId')).toThrow();
    });

    test('fetchFeaturedPlaylists', () => {
      // Use-cases
      expect(
        Uri.Playlist.fetchFeaturedPlaylists({
          territory: Territory.TW,
          limit: 5,
          offset: 10
        })
      ).toMatchSnapshot();

      // Exceptions
      expect(() => Uri.Playlist.fetchFeaturedPlaylists()).toThrow();
    });

    test('fetchNewHitsPlaylists', () => {
      // Use-cases
      expect(
        Uri.Playlist.fetchNewHitsPlaylists({
          territory: Territory.TW,
          limit: 5,
          offset: 10
        })
      ).toMatchSnapshot();

      // Exceptions
      expect(() => Uri.Playlist.fetchNewHitsPlaylists()).toThrow();
    });
  });

  describe('Chart', () => {
    test('listAll', () => {
      // Use-cases
      expect(
        Uri.Chart.listAll({
          territory: Territory.TW,
          limit: 5,
          offset: 10
        })
      ).toMatchSnapshot();

      // Exceptions
      expect(() => Uri.Chart.listAll()).toThrow();
    });
  });

  describe('Search', () => {
    test('query', () => {
      // Use-cases
      expect(
        Uri.Search.query('FAKE QUERY', {
          territory: Territory.TW,
          limit: 5,
          offset: 10
        })
      ).toMatchSnapshot();

      // Exceptions
      expect(() => Uri.Search.query()).toThrow();
    });
  });

  describe('FeaturedPlaylistCategory', () => {
    test('listAll', () => {
      // Use-cases
      expect(
        Uri.FeaturedPlaylistCategory.listAll({
          territory: Territory.TW,
          limit: 5,
          offset: 10
        })
      ).toMatchSnapshot();

      // Exceptions
      expect(() => Uri.FeaturedPlaylistCategory.listAll()).toThrow();
    });
    test('fetch', () => {
      // Use-cases
      expect(
        Uri.FeaturedPlaylistCategory.fetch('fakeCategoryId', {
          territory: Territory.TW,
          limit: 5,
          offset: 10
        })
      ).toMatchSnapshot();

      // Exceptions
      expect(() => Uri.FeaturedPlaylistCategory.fetch()).toThrow();
      expect(() =>
        Uri.FeaturedPlaylistCategory.fetch('fakeCategoryId')
      ).toThrow();
    });
  });

  describe('NewReleaseCategory', () => {
    test('listAll', () => {
      // Use-cases
      expect(
        Uri.NewReleaseCategory.listAll({
          territory: Territory.TW,
          limit: 5,
          offset: 10
        })
      ).toMatchSnapshot();

      // Exceptions
      expect(() => Uri.NewReleaseCategory.listAll()).toThrow();
    });
    test('fetch', () => {
      // Use-cases
      expect(
        Uri.NewReleaseCategory.fetch('fakeCategoryId', {
          territory: Territory.TW,
          limit: 5,
          offset: 10
        })
      ).toMatchSnapshot();

      // Exceptions
      expect(() => Uri.NewReleaseCategory.fetch()).toThrow();
      expect(() => Uri.NewReleaseCategory.fetch('fakeCategoryId')).toThrow();
    });
  });

  describe('MoodStation', () => {
    test('listAll', () => {
      // Use-cases
      expect(
        Uri.MoodStation.listAll({
          territory: Territory.TW,
          limit: 5,
          offset: 10
        })
      ).toMatchSnapshot();

      // Exceptions
      expect(() => Uri.MoodStation.listAll()).toThrow();
    });
    test('fetch', () => {
      // Use-cases
      expect(
        Uri.MoodStation.fetch('fakeStationId', {
          territory: Territory.TW,
          limit: 5,
          offset: 10
        })
      ).toMatchSnapshot();

      // Exceptions
      expect(() => Uri.MoodStation.fetch()).toThrow();
      expect(() => Uri.MoodStation.fetch('fakeStationId')).toThrow();
    });
  });

  describe('GenreStation', () => {
    test('listAll', () => {
      // Use-cases
      expect(
        Uri.GenreStation.listAll({
          territory: Territory.TW,
          limit: 5,
          offset: 10
        })
      ).toMatchSnapshot();

      // Exceptions
      expect(() => Uri.GenreStation.listAll()).toThrow();
    });
    test('fetch', () => {
      // Use-cases
      expect(
        Uri.GenreStation.fetch('fakeStationId', {
          territory: Territory.TW,
          limit: 5,
          offset: 10
        })
      ).toMatchSnapshot();

      // Exceptions
      expect(() => Uri.GenreStation.fetch()).toThrow();
      expect(() => Uri.GenreStation.fetch('fakeStationId')).toThrow();
    });
  });
});
