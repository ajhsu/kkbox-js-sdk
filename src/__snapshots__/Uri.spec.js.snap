// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uri Album fetchMetadata 1`] = `"https://api.kkbox.com/v1.1/albums/fakeAlbumId?territory=TW"`;

exports[`Uri Album fetchTracks 1`] = `"https://api.kkbox.com/v1.1/albums/fakeAlbumId/tracks?territory=TW&limit=5&offset=10"`;

exports[`Uri Artist fetchAlbums 1`] = `"https://api.kkbox.com/v1.1/artists/fakeArtistId/albums?territory=TW&limit=5&offset=10"`;

exports[`Uri Artist fetchMetadata 1`] = `"https://api.kkbox.com/v1.1/artists/fakeArtistId?territory=TW"`;

exports[`Uri Artist fetchRelatedArtists 1`] = `"https://api.kkbox.com/v1.1/artists/fakeArtistId/related-artists?territory=TW&limit=5&offset=10"`;

exports[`Uri Artist fetchTopTracks 1`] = `"https://api.kkbox.com/v1.1/artists/fakeArtistId/top-tracks?territory=TW&limit=5&offset=10"`;

exports[`Uri Auth fetchAccessToken 1`] = `"https://account.kkbox.com/oauth2/token/"`;

exports[`Uri Chart listAll 1`] = `"https://api.kkbox.com/v1.1/charts?territory=TW&limit=5&offset=10"`;

exports[`Uri FeaturedPlaylistCategory fetch 1`] = `"https://api.kkbox.com/v1.1/featured-playlist-categories/fakeCategoryId?territory=TW&limit=5&offset=10"`;

exports[`Uri FeaturedPlaylistCategory listAll 1`] = `"https://api.kkbox.com/v1.1/featured-playlist-categories?territory=TW&limit=5&offset=10"`;

exports[`Uri GenreStation fetch 1`] = `"https://api.kkbox.com/v1.1/genre-stations/fakeStationId?territory=TW&limit=5&offset=10"`;

exports[`Uri GenreStation listAll 1`] = `"https://api.kkbox.com/v1.1/genre-stations?territory=TW&limit=5&offset=10"`;

exports[`Uri MoodStation fetch 1`] = `"https://api.kkbox.com/v1.1/mood-stations/fakeStationId?territory=TW&limit=5&offset=10"`;

exports[`Uri MoodStation listAll 1`] = `"https://api.kkbox.com/v1.1/mood-stations?territory=TW&limit=5&offset=10"`;

exports[`Uri NewReleaseCategory fetch 1`] = `"https://api.kkbox.com/v1.1/new-release-categories/fakeCategoryId?territory=TW&limit=5&offset=10"`;

exports[`Uri NewReleaseCategory listAll 1`] = `"https://api.kkbox.com/v1.1/new-release-categories?territory=TW&limit=5&offset=10"`;

exports[`Uri Playlist fetch 1`] = `"https://api.kkbox.com/v1.1/shared-playlists/fakePlaylistId?territory=TW"`;

exports[`Uri Playlist fetchFeaturedPlaylists 1`] = `"https://api.kkbox.com/v1.1/featured-playlists?territory=TW&limit=5&offset=10"`;

exports[`Uri Playlist fetchNewHitsPlaylists 1`] = `"https://api.kkbox.com/v1.1/new-hits-playlists?territory=TW&limit=5&offset=10"`;

exports[`Uri Search query 1`] = `"https://api.kkbox.com/v1.1/search?q=FAKE%20QUERY&territory=TW&limit=5&offset=10"`;

exports[`Uri Track fetch 1`] = `"https://api.kkbox.com/v1.1/tracks/fakeTrackId?territory=TW"`;
